src/fondant/component_spec.py

```diff
@@ -230,85 +229,159 @@ class KubeflowComponentSpec:
+            },
+            "parameters": {
+                "component_spec": {
                     "description": "The component specification as a dictionary",
-                    "type": "JsonObject",
-                    "default": "None",
+                    "defaultValue": {},
+                    "isOptional": True,
+                    "parameterType": "STRUCT",
                 },
-                {
-                    "name": "input_partition_rows",
-                    "description": "The number of rows to load per partition. Set to override the"
-                    " automatic partitioning",
-                    "type": "String",
-                    "default": "None",
+                "input_partition_rows": {
+                    "description": "The number of rows to load per partition."
+                    + " Set to override the automatic partitioning",
+                    "isOptional": True,
+                    "parameterType": "STRING",
+                    "defaultValue": "None",
                 },
-                {
-                    "name": "cache",
+                "cache": {
+                    "parameterType": "BOOLEAN",
                     "description": "Set to False to disable caching, True by default.",
-                    "type": "Boolean",
-                    "default": "True",
+                    "defaultValue": True,
+                    "isOptional": True,
                 },
-                *(
-                    {
-                        "name": arg.name,
-                        "description": arg.description,
-                        "type": python2kubeflow_type[arg.type],
-                        **({"default": arg.default} if arg.default is not None else {}),
-                    }
-                    for arg in fondant_component.args.values()
-                ),
-            ],
-            "outputs": [
-                {
-                    "name": "output_manifest_path",
+                "metadata": {
+                    "description": "Metadata arguments containing the run id and base path",
+                    "parameterType": "STRING",
+                },
+                **cls.convert_arguments(fondant_component),
+            },
+        }
+
+        cleaned_component_name = fondant_component.name.replace("-", "_").replace(
+            " ",
+            "_",
+        )
+        output_definitions = {
+            "artifacts": {
+                "output_manifest_path": {
+                    "artifactType": {
+                        "schemaTitle": "system.Artifact",
+                        "schemaVersion": "0.0.1",
+                    },
                     "description": "Path to the output manifest",
-                    "type": "String",
                 },
-            ],
-            "implementation": {
-                "container": {
-                    "image": fondant_component.image,
-                    "command": [
```

Comment:
I think this is the command that you replaced, we're using this now for the components based on this PR https://github.com/ml6team/fondant/commits/main?before=82b5131ac8e72c620917d65c31c80b0b903e4d83+35&branch=main&qualified_name=refs%2Fheads%2Fmain

Can you replace and check if all still runs fine?

Reply:
@mrchtr

Reply:
updated